src/utils/utilHandlers.ts

```diff
@@ -314,15 +297,68 @@ export async function handleDsCreation(
     const collection = Collection.create({
       id: collectionId,
       networkId: chainId,
-      contract_address: casedAddress,
+      contract_address: casedContractAddress,
       created_block: blockNumber,
       created_timestamp: timestamp,
-      creator_address: creatorAddress.toLowerCase(),
+      creator_address: casedCreatorAddress,
       total_supply: totalSupplyResult.toBigInt(),
       name,
       symbol,
-      contract_type: ContractType.ERC721,
+      contract_type: contractType,
     });
     await collection.save();
   }
+
+  if (contractType === ContractType.ERC1155) {
+    logger.info(`is erc1155, address=${casedContractAddress}`);
+
+    const collection = Collection.create({
+      id: collectionId,
+      networkId: chainId,
+      contract_address: casedContractAddress,
+      created_block: blockNumber,
+      created_timestamp: timestamp,
+      creator_address: casedCreatorAddress,
+      total_supply: BigInt(0),
+      contract_type: contractType,
+    });
+    await collection.save();
+  }
+}
+
+export async function collectionController(event: {
+  address: string;
+  blockNumber: number;
+  block: { timestamp: bigint };
+  transaction: { from: string };
+}): Promise<void> {
+  // check if apart from blacklist table
+  const casedContractAddress = event.address.toLowerCase();
+
+  await handleNetwork(chainId);
+
+  const collectionId = getCollectionId(chainId, casedContractAddress);
+  const collection = await Collection.get(collectionId);
+  if (collection) {
+    logger.info(`Skipping collection: ${collectionId} as it exists already`);
+    return;
+  }
+
+  // Check if desired interface
+  const [isErc1155, isErc721] = await interfaceCheck(event.address);
+
+  if (!isErc721 && !isErc1155) {
+    throw new Error('Contract does not implement erc165');
```

Comment:
I don't think this message is right. 

If it doesn't implement ERC165 then interfaceCheck would throw.